analyses/fusion_filtering/02-fusion-filtering.R

```diff
@@ -0,0 +1,237 @@
+              help="Filtered fusion calls (prefix for _putDriver.RDS and _filtFusion.RDS)")
+)
+
+
+# get command line options, if help option encountered print help and exit,
+
+opt <- parse_args(OptionParser(option_list=option_list))
+
+standardFusionFile<-opt$standardFusionFile
+expressionMatrix<-opt$expressionMatrix
+reathroughFilter<-opt$reathroughFilter
+expressionFilter<-opt$expressionFilter
+artifactFilter<-opt$artifactFilter
+junctionReadCountFilter<-opt$junctionReadCountFilter
+spanningFragCountFilter<-opt$spanningFragCountFilter
+putativeDriverGeneList<-opt$putativeDriverGeneList
+filterGeneList<-opt$filterGeneList
+readingFrameFilter<-opt$readingFrameFilter
+
+# read standardized fusion calls
+standardFusioncalls<-read.delim(standardFusionFile,stringsAsFactors=FALSE)
+# to obtain geneA and geneB for gene search below
+standardFusioncalls <- cbind(standardFusioncalls, colsplit(standardFusioncalls$FusionName, pattern = '--', names = c("GeneA","GeneB")))
+standardFusioncalls <- cbind(standardFusioncalls, colsplit(standardFusioncalls$GeneA, pattern = '/', names = c("Gene1A","Gene2A")))
+standardFusioncalls <- cbind(standardFusioncalls, colsplit(standardFusioncalls$GeneB, pattern = '/', names = c("Gene1B","Gene2B")))
+
+
+# gather gene and fusion list from reference folder
+# tab delimited gene of interest merge with filename for putativeDriver
+putDriverReferenceDataTab<- tibble(filename = c(paste0(opt$referenceFolder,unlist(strsplit(putativeDriverGeneList,","))))) %>% 
+  # create a data frame
+  # holding the file names
+  mutate(file_contents = map(filename,read.delim,stringsAsFactor=FALSE,sep="\t")) # a new data column
+putDriverReferenceDataTab<-tidyr::unnest(putDriverReferenceDataTab) %>% as.data.frame()
+
+#gather column 1 as GeneName 2 as FusionName and 3 as source
+putDriverReferenceDataTab<-data.frame("GeneName"=c(putDriverReferenceDataTab$symbol[!is.na(putDriverReferenceDataTab$symbol)],putDriverReferenceDataTab$GeneSymbol[!is.na(putDriverReferenceDataTab$GeneSymbol)],putDriverReferenceDataTab$Gene.Symbol[!is.na(putDriverReferenceDataTab$Gene.Symbol)],putDriverReferenceDataTab$Gene_A[!is.na(putDriverReferenceDataTab$Gene_A)],putDriverReferenceDataTab$Gene_B[!is.na(putDriverReferenceDataTab$Gene_B)]),"FusionName"=NA,"source"=c(putDriverReferenceDataTab$filename[!is.na(putDriverReferenceDataTab$symbol)],putDriverReferenceDataTab$filename[!is.na(putDriverReferenceDataTab$GeneSymbol)],putDriverReferenceDataTab$filename[!is.na(putDriverReferenceDataTab$Gene.Symbol)],putDriverReferenceDataTab$filename[!is.na(putDriverReferenceDataTab$Gene_A)],putDriverReferenceDataTab$filename[!is.na(putDriverReferenceDataTab$Gene_B)]),stringsAsFactors = FALSE)
+
+
+# tab delimited gene of interest merge with filename for filter fusion with genes other than putative driver gene list
+filtFusionrReferenceDataTab<- tibble(filename = c(paste0(opt$referenceFolder,unlist(strsplit(filterGeneList,","))))) %>% 
```

Comment:
Well we want to keep putative driver list as genes that are known to be cancer specific as in oncogenic/cosmic census/pan-cancer fusion genes/tumor suppressor . And filtered fusion is other genes of interest like TF and kinases. @jharenza can you provide more information if moving forward we want to keep these steps separate?

Reply:
@jaclyn-taroni @kgaonkar6 - rationale for this was that after artifact filtering, the genes are simply annotated, but for `putative drivers`, we did additional filtering for gene partners which are known TSG/oncogenes/in COSMIC. @kgaonkar6 are we filtering the annotated list for TF/Kinases, as you suggest? I think we should probably not filter this, in case someone simply wants to annotate all results and do their own filtering.

Reply:
Ok cool, I'll add annotation as the first step in this script; then filtering artifacts save as annotated QC fusion calls and then save putdriver genes as a separate file as we discussed in person :)

Reply:
Updated to annotating all calls and no filtering on gene list in this general filtering step